File: PregnancyAppBackend/Dtos/Web/MedicalHistory/MedicalHistoryDto.cs
using PregnancyAppBackend.Enums.MedicalHistory;

namespace PregnancyAppBackend.Dtos.Web.MedicalHistory;

public class MedicalHistoryDto
{
    public int Weight { get; set; }
    public int Height { get; set; }
    public BloodGroup BloodGroup { get; set; }
    public RhesusFactor RhesusFactor { get; set; }
    public string BloodPressure { get; set; } = null!;
    public Thermometer Thermometer { get; set; }
    public int PregnancyAmount { get; set; }
    public int? AbortionAmount { get; set; }
    public int? MiscarriageAmount { get; set; }
    public int? PrematureBirthAmount { get; set; }
    public BirthType PreviousBirthType { get; set; }
    public string? GynecologicalDiseases { get; set; }
    public string? SomaticDiseases { get; set; }
    public string? UndergoneOperations { get; set; }
    public string? AllergicReactions { get; set; }
    public List<HereditaryDisease> HereditaryDiseases { get; set; } = new();
    public bool IsSmoking { get; set; }
    public bool IsConsumingAlcohol { get; set; }
    public CovidStatus EnduredCovid { get; set; }
}

File: PregnancyAppFrontend/src/app/dtos/medical-history/medical-history-dto.ts
import {BloodGroup} from "../../enums/medical-history/blood-group";
import {RhesusFactor} from "../../enums/medical-history/rhesus-factor";
import {Thermometer} from "../../enums/medical-history/thermometer";
import {BirthType} from "../../enums/medical-history/birth-type";
import {CovidStatus} from "../../enums/medical-history/covid-status";
import {HereditaryDisease} from "../../enums/medical-history/hereditary-disease";

export interface MedicalHistoryDto {
  weight: number;
  height: number;
  bloodGroup: BloodGroup;
  rhesusFactor: RhesusFactor;
  bloodPressure: string;
  thermometer: Thermometer;
  pregnancyAmount: number;
  abortionAmount?: number;
  miscarriageAmount?: number;
  prematureBirthAmount?: number;
  previousBirthType: BirthType;
  gynecologicalDiseases?: string;
  somaticDiseases?: string;
  undergoneOperations?: string;
  allergicReactions?: string;
  hereditaryDiseases: Array<HereditaryDisease>;
  isSmoking: boolean;
  isConsumingAlcohol: boolean;
  enduredCovid: CovidStatus;
}
